scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/converter/EscapeSpaceURIConverter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.common.job.factories.spi.model.converter;
+
+import java.net.URI;
+
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.exceptions.ConversionException;
+
+
+public class EscapeSpaceURIConverter extends URIConverter {
+
+    public final String encodedSpace = "%20";
+
+    @Override
+    public URI convert(String parameterValue) throws ConversionException {
+        String escapedValue = parameterValue.replace(" ", encodedSpace);
```

Comment:
This looks valid to circumvent issue from space character. But what happen when other URL-unfriendly are met ? Shouldn't you use `URLEncoder.encode()` to address them all at once ?

Reply:
Personally I think it's more safe to disallow those characters instead of encoding them. To handle all these characters also requires work in frontend path encoding and backend vfs resolveFile decoding. Currently I don't feel it's really needed. 

@fviale What do you think ?

Reply:
By implementing ourself an encoder we take the risk to miss unsupported characters.

Reply:
@tao-xinxiu I think we need to find the most robust solution

Reply:
For example this seems to work for relative paths:

path = new File( "ab cd/toto titi/file.txt").toURI().toURL().toString()
root = new File("").toURI().toURL().toString()
println path.replace(root, "")

Reply:
are we ever going to have absolute paths here ?

Reply:
I think it should always be relative path here, as the user may (or should) have no knowledge about the server dataspace root path.

Reply:
After verification, escaping special characters is already handled in vfs, so stopping convert variable value to URI fix the pb.